pkg/controller/certificates/cleaner/cleaner.go

```diff
@@ -158,6 +159,19 @@ func isDeniedPastDeadline(csr *capi.CertificateSigningRequest) bool {
 	return false
 }
 
+// isFailedPastDeadline checks if the certificate has a Failed status and the
+// creation time of the CSR is passed the deadline that pending requests are
+// maintained for.
+func isFailedPastDeadline(csr *capi.CertificateSigningRequest) bool {
```

Comment:
Despite this being quite simple, might be worth including a unit test here given there's already plumbing for it? https://github.com/kubernetes/kubernetes/blob/dcabb7e6a81e7e60a43a82a380bf01fd0604b8f7/pkg/controller/certificates/cleaner/cleaner_test.go

Unless there's an integration test I've missed 😄

Reply:
added a unit test